### PR TITLE
Tooling API CRUD fix (#842)

### PR DESCRIPTION
**Pull request by superkensington**
Added missing method to Tooling API class from base Connection class and dependent property. Also added some basic CRUD tests for the Tooling API since none existed, modeled after the Connection CRUD tests. Fixes jsforce/jsforce#842

### CODE REVIEW

File: src/api/tooling.ts [hunk 1]
```diff
@@ -181,6 +181,7 @@ export type CompletionsResult = {
 const {
   query,
   queryMore,
+    _ensureVersion,
```
Comment (kyungkimsf) [inline src/api/tooling.ts:184]: nit: indentation seems to be off for the added lines of code in this file

Comment (mosel-slalom) [reply src/api/tooling.ts:184]: @kyungkimsf is it possible to get this merged soon?

Comment (superkensington) [reply src/api/tooling.ts]: > nit: indentation seems to be off for the added lines of code in this file

✅ Fixed
